src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,134 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.TaskList;
+import duke.task.ToDo;
+
+import java.util.Scanner;
+
+public class Duke {
+    public static final String COMMAND_BYE = "bye";
+    public static final String COMMAND_LIST = "list";
+    public static final String COMMAND_TODO = "todo";
+    public static final String COMMAND_DEADLINE = "deadline";
+    public static final String COMMAND_EVENT = "event";
+    public static final String COMMAND_DONE = "done";
+
+    public static void printHorizontalLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void printGreetingMessage() {
+        printHorizontalLine();
+        System.out.println("Hello! I'm duke.Duke");
+        System.out.println("What can I do for you?");
+        printHorizontalLine();
+    }
+
+    public static void printExitMessage() {
+        printHorizontalLine();
+        System.out.println("Bye. Hope to see you again soon!");
+        printHorizontalLine();
+    }
+
+    public static String getCommand(String line) {
+        return line.split(" ")[0];
+    }
+
+    public static String getTaskDescription(String command, String line) throws DukeException {
+        if (line.split(command).length < 2 || line.split(command)[1].equals(" ")) {
+            throw new DukeException("☹ OOPS!!! Missing duke.task description!");
+        } else {
+            switch (command) {
+            case COMMAND_TODO:
+                return line.substring(line.indexOf(' ') + 1);
+            case COMMAND_DEADLINE:
+            case COMMAND_EVENT:
+                if (!line.contains("/by ") && command.equals(COMMAND_DEADLINE)) {
```

Comment:
Might this be better/easier to follow if not a fall-through? Like 

`case COMMAND_DEADLINE: if (!line.contains("/by") { throw new DukeException("☹ OOPS!!! Missing or incorrect /by statement"); }`,

and similar for the COMMAND_EVENT, then followed by the fall-through case of

`return line.substring(line.indexOf(' ') + 1, line.indexOf('/') - 1);`

Otherwise, super good code and readable, easy to follow, and very logical.

Reply:
Good idea! I will implement that!